src/KafkaFlow.Retry/Durable/Polling/QueueTracker.cs

```diff
@@ -1,130 +1,171 @@
+                            CronExpression = jobDataProvider.PollingDefinition.CronExpression
                         });
+
+                    continue;
                 }
 
-                if (!this.IsSchedulerActive)
-                {
-                    this.scheduler
-                        .Start(cancellationToken)
-                        .GetAwaiter()
-                        .GetResult();
+                this.ScheduleJob(jobDataProvider, cancellationToken);
+            }
+        }
 
-                    this.logHandler.Info(
-                        "PollingJob Scheduler Started",
-                        new
-                        {
-                            PollingId = this.retryDurablePollingDefinition.Id,
-                            CronExpression = this.retryDurablePollingDefinition.CronExpression
-                        });
+        internal void UnscheduleJobs(CancellationToken cancellationToken = default)
+        {
+            foreach (var jobDataProvider in this.jobDataProviders)
+            {
+                if (!jobDataProvider.PollingDefinition.Enabled)
+                {
+                    continue;
                 }
 
+                var trigger = this.triggerProvider.GetPollingTrigger(this.schedulerId, jobDataProvider.PollingDefinition);
+
+                this.logHandler.Info(
+                    "PollingJob unscheduler started",
+                    new
+                    {
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        TriggerKey = trigger.Key.ToString()
+                    });
+
+                var unscheduledJob = this.scheduler
+                    .UnscheduleJob(trigger.Key)
+                    .GetAwaiter()
+                    .GetResult();
+
+                this.logHandler.Info("PollingJob unscheduler finished",
+                    new
+                    {
+                        SchedulerId = this.schedulerId,
+                        PollingJobType = jobDataProvider.PollingDefinition.PollingJobType.ToString(),
+                        TriggerKey = trigger.Key.ToString(),
+                        UnscheduledJob = unscheduledJob.ToString()
+                    });
+            }
+        }
+
+        private void ScheduleJob(IJobDataProvider jobDataProvider, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var job = jobDataProvider.GetPollingJobDetail();
+                var trigger = this.triggerProvider.GetPollingTrigger(this.schedulerId, jobDataProvider.PollingDefinition);
+
                 var scheduledJob = this.scheduler
-                    .ScheduleJob(this.job, this.trigger, cancellationToken)
+                    .ScheduleJob(job, trigger, cancellationToken)
                     .GetAwaiter()
                     .GetResult();
```

Comment:
Perhaps we can make this statement and the method async?

Reply:
same response from above